streampark-flink/streampark-flink-client/streampark-flink-client-core/src/main/scala/org/apache/streampark/flink/client/trait/FlinkClientTrait.scala

```diff
@@ -539,10 +539,6 @@ trait FlinkClientTrait extends Logger {
       case (false, false) =>
         client.cancel(jobID).get()
         null
-      case (true, false) =>
-        clientWrapper
-          .cancelWithSavepoint(jobID, savePointDir, cancelRequest.nativeFormat)
```

Comment:
We need to know the difference between `canceWithSavepoint` and `stopWithSavepoint` and in what scenarios they are used

Reply:
I'm not sure if it will cause a bug in different modes (yarn-perjob | yarn-session), can you test and confirm?

Reply:
According to Flink documentation,using `cancel` command to stop a job with savepoint is deprecated.Use the `stop` action instead. It is a more graceful way of stopping a running streaming job as the stop flows from source to sink. When the user requests to stop a job, all sources will be requested to send the last checkpoint barrier that will trigger a savepoint, and after the successful completion of that savepoint, they will finish by calling their `cancel()` method. 

I have tested yarn-perjob and yarn-session, they can be stopped normally.

Reply:
I tested this case in `flink 1.17`,this is stop command  related document https://nightlies.apache.org/flink/flink-docs-master/docs/deployment/cli/#terminating-a-job

Reply:
Thanks for your clarification, LGTM